internal/errmeta/errmeta.go

```diff
@@ -0,0 +1,29 @@
+package errmeta
+
+func New(err error, meta map[string]string) error {
+	return &ErrMeta{
+		err:      err,
+		metadata: meta,
+	}
+}
+
+// ErrMeta represents an error with attached metadata for additional runtime context
+type ErrMeta struct {
```

Comment:
this still feels like a bit of overkill, your package shouldn't need to define a new errors package. and ideally it should be able to hand off logging by returning errors back up to the client.

what about creating specific error types for the error cases you have? so, instead of this package you could have
```
type PauseError struct {
    OriginalErr error
    RunID string
    ForeignID string
}
```
```
maybePause(ctx, pauseAfterErrCount, w.errorCounter, err, processName, run, w.logger)
		if err != nil {
			return PauseError{OriginalErr: err, RunID: record.RunID, ForeignID: record.ForeignID}
		})
```
then a client can use errors.As to process the details if it needs to?

Reply:
I have gone and remove the errmeta package, not sure why I didn't think of this sooner, is to just use `fmt.Errorf` to provide context. This pushing for a more consistent use of "opaque" errors. This satisfies my aims and avoids having a dedicated package to pass meta to the logger.